### DISCUSSION TITLE
2023-24: 漩涡事件反思录

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#406] [Announcements]
# 2023-24: 漩涡事件反思录

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/reports/2023-24/

**Comment by Endle** [comment]
> 外部的贡献者已经很难直接与对项目有实际影响力的人取得联系。与此同时，公司内部的维护者也更倾向于使用方便快捷的内部沟通渠道：飞书私聊 ping 一下显然要比通过 Github 沟通方便得多


对于这个问题，有一个可能过于理想化的方案。是否可以考虑，将项目的开发流程，从内部移动到公开平台上？即便是几名公司雇员相互 assign ticket，也将其透明化。项目设计的争论，在不涉及机密时，也尽量对社区公开

**Reply by Xuanwo** [reply]
PingCAP 曾经是这样：所有的讨论都在 Slack 上，绝大多数技术讨论完全对外公开（因为当时内部沟通使用企业微信太难用了）。后来切换到了飞书之后大家就慢慢的都开始用飞书了（

**Comment by noirgif** [comment]
看到标题我还以为是老涡捅了一个什么篓子…… 还好还好

我觉得就算是飞书多方便，作为一个开源项目，有即时的沟通渠道能拉近跟贡献者的距离，而且让交流更高效。

另外，如果TiDB通常都是作为一个整体使用的话（TiDB有除了TiKV之外的后端吗？），其实合并起来作为一个monorepo更能节省维护的空间？像老涡之前博客写的一样。

**Comment by manuzhang** [comment]
既然有代码查重工具，也应该有 PR 查重工具吧，人总是不可靠的（中性）。

**Comment by guojidan** [comment]
这种事情太常见了，我的好多PR或者issue无人回应，越成熟的项目想要加入越难，除非没有商业公司的纯社区项目

**Comment by lewiszlw** [comment]
我这周刚经历了这种事😂，给清华 rCore-Tutorial-v3 提了个 bugfix pr https://github.com/rcore-os/rCore-Tutorial-v3/pull/154 ，结果一直没人合并，然后清华学生晚了接近一个月，提了个一模一样的bugfix pr https://github.com/rcore-os/rCore-Tutorial-v3/pull/155 被合并进去了，大概率可能还是无心之举吧 :)
